htlcswitch/switch.go

```diff
@@ -1163,12 +1169,14 @@ func (s *Switch) handlePacketForward(packet *htlcPacket) error {
 // The ciphertext will be derived from the failure message proivded by context.
 // This method returns the failErr if all other steps complete successfully.
 func (s *Switch) failAddPacket(packet *htlcPacket,
-	failure lnwire.FailureMessage, failErr error) error {
+	failure *LinkError, failErr error) error {
```

Comment:
can just take the `FailureMessage` as before?

Reply:
Ah yes, getting ahead of myself for htlcNotifier things :D